engine.py

```diff
@@ -301,8 +301,14 @@ def _emit_log_message(self, handler, record):
             record.message
         )
 
-        sys.stdout.write(msg_str)
-        sys.stdout.flush()
+        try:
```

Comment:
I think this is because our console redirect classes don't implement `flush()`. I think the proper solution here is to fix the console redirectors.